sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelUtils.java

```diff
@@ -61,8 +75,17 @@ public static void setIOFactory(String scheme, IOChannelFactory factory) {
    * to provide, e.g., credentials for GCS.
    */
   public static void registerStandardIOFactories(PipelineOptions options) {
-    setIOFactory("gs", new GcsIOChannelFactory(options.as(GcsOptions.class)));
-    setIOFactory("file", new FileIOChannelFactory());
+    setIOFactory("gs", GcsIOChannelFactory.fromOptions(options));
```

Comment:
Shouldn't this method now always throw since setIOFactory will conflict with registrar entries now provided by the `@AutoService` annotation or is the `@AutoService` stuff not working for you?

Also, we have several places in our code where we force registration of standard IO factories. These additional places where we make these calls will always now through.